File: solutions/javascript/word-count/2/word-count.js
"use strict";
var Words = (function() {
  function Words(wordString) {
    this.wordString = wordString.toLowerCase();
    this.count = this.counts();
  }
  Words.prototype.counts = function() {
    var counts = this.wordString.match(/\b[a-z0-9]+\b/gi).
      reduce(function(collection, word) {
        collection[word] = (collection[word] || 0) + 1;
        return collection;
      }, {});
    return counts;
  }
  return Words;
})();

module.exports = Words;

File: solutions/elixir/word-count/2/word-count.exs
defmodule Words do
  def count(phrase), do: normalize_words(phrase) |> count_words

  defp normalize_words(phrase), do: Regex.scan(%r/\w+/, String.downcase(phrase))

  defp count_words(words) do
    Enum.reduce words, HashDict.new, fn word, acc -> HashDict.update(acc, word, 1, fn val -> val + 1 end) end
  end
end